ecommerce-frontend/src/Routes.js

```diff
@@ -15,7 +17,8 @@ const Routes = () => {
         <Route path="/" exact component={Home} />
         <Route path="/signin" exact component={SignIn} />
         <Route path="/signup" exact component={SignUp} />
-        <PrivateRoute path="/dashboard" exact component = {Dashboard}/>
+        <PrivateRoute path="/user/dashboard" exact component={Dashboard} />
```

Comment:
Just create two components `PrivateRoute` and `PublicRoute`. Modify your logic accordingly.